File: CODE_FILES/TASK2/GO/Task2.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strings"
    "time"
)

func main() {
    start := time.Now()
    file, _ := os.Open("input.txt")
    defer file.Close()

    scanner := bufio.NewScanner(file)
    count := 0

    for scanner.Scan() {
        if strings.Contains(scanner.Text(), "data") {
            count++
        }
    }

    fmt.Println("Lines containing 'data':", count)
    fmt.Println("Time taken:", time.Since(start).Milliseconds(), "ms")

    // Adding time to see it in Process Explorer
    time.Sleep(3 * time.Second)
}

File: CODE_FILES/TASK2/JAVA/Task2.java
import java.io.*;
import java.nio.file.*;

public class Task2 {
    public static void main(String[] args) throws IOException {
        long start = System.currentTimeMillis();
        int count = 0;

        try (BufferedReader reader = Files.newBufferedReader(Paths.get("input.txt"))) {
            String line;
            while ((line = reader.readLine()) != null) {
                if (line.contains("data")) {
                    count++;
                }
            }
        }

        long end = System.currentTimeMillis();
        System.out.println("Lines containing 'data': " + count);
        System.out.println("Time taken: " + (end - start) + " ms");

	try {
    		Thread.sleep(3000); // sleep for 3 seconds
	} catch (InterruptedException e) {
    		e.printStackTrace();
	}
    }
}
